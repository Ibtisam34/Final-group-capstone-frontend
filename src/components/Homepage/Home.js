import React from 'react';
import { Outlet } from 'react-router';
import { Container } from 'react-bootstrap';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import PropTypes from 'prop-types';
import SideBar from '../Navbar/SideBar';

function Home({ isLoggedIn }) {
  return (
    <Container fluid>
      <Row className="homepage-container bg-white">
        <Col md={2} lg={2} className="nav-panel d-flex flex-column justify-content-between bg-light px-0">
          <SideBar isLoggedIn={isLoggedIn} />
        </Col>
        <Col md={10} lg={10} className="content-panel">
          <Outlet />
        </Col>
      </Row>
    </Container>
  );
}

Home.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
};

export default Home;
