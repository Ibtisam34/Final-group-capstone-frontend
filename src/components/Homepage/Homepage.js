import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Button } from 'react-bootstrap';
import PropTypes from 'prop-types';
import Slider from './SliderHomepage';

function Homepage({ isLoggedIn }) {
  const navigate = useNavigate();

  const handleLogout = () => {
    const keys = ['isLoggedIn', 'userId'];

    keys.forEach((key) => localStorage.removeItem(key));

    navigate('/');
    window.location.reload();
  };

  return (
    <div>
      <div className="homepage-header-container">
        <h1 id="h1">Top-rated Doctors</h1>
        <p>Find and book an appointment</p>
      </div>

      <Slider isLoggedIn={isLoggedIn} />

      {isLoggedIn && (
        <Button
          variant="secondary"
          className="logout-btn"
          onClick={() => handleLogout()}
        >
          Logout
        </Button>
      )}
    </div>
  );
}

Homepage.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
};

export default Homepage;
