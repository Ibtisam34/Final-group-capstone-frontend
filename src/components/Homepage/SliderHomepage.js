import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Carousel from 'react-multi-carousel';
import 'react-multi-carousel/lib/styles.css';
import PropTypes from 'prop-types';
import Detail from '../Details';
import { fetchDoctors } from '../../redux/doctors/doctorsReducer';

function Slider({ isLoggedIn }) {
  const responsive = {
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 3,
      slidesToSlide: 3, // optional, default to 1.
    },
    tablet: {
      breakpoint: { max: 1024, min: 454 },
      items: 2,
      slidesToSlide: 2, // optional, default to 1.
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 1,
      slidesToSlide: 1, // optional, default to 1.
    },
  };
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchDoctors());
  }, [dispatch]);

  const [selectedDoctor, setSelectedDoctor] = useState(null);
  const doctorObj = useSelector(({ doctorsReducer }) => doctorsReducer);

  const handleSelected = (e) => setSelectedDoctor(e);

  return (
    <div>
      {!selectedDoctor ? (
        <Carousel
          arrows
          renderButtonGroupOutside
          swipeable
          draggable={false}
          responsive={responsive}
          infinite
          autoPlaySpeed={8000}
          keyBoardControl
          customTransition="all .5"
          transitionDuration={500}
          containerClass="carousel-container"
          dotListClass="custom-dot-list-style"
          itemClass="carousel-item-padding-40-px"
        >
          {doctorObj.map((doctor) => (
            <div key={doctor.id} className="carousel-img-container" role="link" tabIndex="0" onKeyDown={() => null} onClick={() => handleSelected(doctor)}>
              <img src={doctor.image} alt="doctor" />
              <div className="carousel-description">
                <p>
                  <strong>{doctor.name}</strong>
                  {' '}
                </p>
                <span>{doctor.specialization}</span>
                <p>{doctor.availability}</p>
              </div>
            </div>
          ))}
        </Carousel>
      ) : (
        <Detail doctor={selectedDoctor} isLoggedIn={isLoggedIn} />
      )}
    </div>
  );
}

Slider.propTypes = {
  isLoggedIn: PropTypes.bool.isRequired,
};

export default Slider;
