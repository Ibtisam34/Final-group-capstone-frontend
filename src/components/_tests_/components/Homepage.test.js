import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/configureStore';
import Homepage from '../../Homepage/Homepage';

describe('Home page', () => {
  it('renders text in home page', () => {
    render(
      <Provider store={store}>
        <Router>
          <Homepage />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('Top-rated Doctors')).toBeInTheDocument();
  });

  it('renders homapage correctly', () => {
    const tree = render(
        <Provider store={store}>
            <Router>
                <Homepage />
            </Router>
        </Provider>);
        expect(tree).toMatchSnapshot();
  })
});
