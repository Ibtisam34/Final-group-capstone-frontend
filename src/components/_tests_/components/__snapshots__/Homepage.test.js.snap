// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders homapage correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="homepage-header-container"
        >
          <h1
            id="h1"
          >
            Top-rated Doctors
          </h1>
          <p>
            Find and book an appointment
          </p>
        </div>
        <div>
          <div
            class="react-multi-carousel-list carousel-container "
            dir="ltr"
          >
            <ul
              class="react-multi-carousel-track "
              style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="homepage-header-container"
      >
        <h1
          id="h1"
        >
          Top-rated Doctors
        </h1>
        <p>
          Find and book an appointment
        </p>
      </div>
      <div>
        <div
          class="react-multi-carousel-list carousel-container "
          dir="ltr"
        >
          <ul
            class="react-multi-carousel-track "
            style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
